jujugui/static/gui/src/app/components/basic-table/basic-table.js

```diff
@@ -0,0 +1,112 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+/** Basic table React component used to display data in a table structure. */
+class BasicTable extends React.Component {
+  constructor() {
+    super();
+    this.columnClasses = [
```

Comment:
Vanilla will begin to support CSS grids soon and in by proxy so will all apps and sites across the company. It is very helpful to have a central place to develop a framework which is rolled out across all the products.